src/compiler/scala/tools/nsc/symtab/classfile/ClassfileParser.scala

```diff
@@ -1313,6 +1261,76 @@ abstract class ClassfileParser {
+            val paramsNoOuter = innerClasses getEntry currentClass match {
+              case Some(entry) if !entry.jflags.isStatic =>
+                /* About `clazz.owner.hasPackageFlag` below: scala/bug#5957
+                 * For every nested java class A$B, there are two symbols in the scala compiler.
+                 *  1. created by SymbolLoader, because of the existence of the A$B.class file, owner: package
+                 *  2. created by ClassfileParser of A when reading the inner classes, owner: A
+                 * If symbol 1 gets completed (e.g. because the compiled source mentions `A$B`, not `A#B`), the
+                 * ClassfileParser for 1 executes, and clazz.owner is the package.
+                 */
+                assert(params.head.tpe.typeSymbol == clazz.owner || clazz.owner.hasPackageFlag, params.head.tpe.typeSymbol + ": " + clazz.owner)
+                params.tail
+              case _ =>
+                params
+            }
+            val newParams = paramsNoOuter match {
+              case (init :+ tail) if jflags.isSynthetic =>
+                // scala/bug#7455 strip trailing dummy argument ("access constructor tag") from synthetic constructors which
+                // are added when an inner class needs to access a private constructor.
+                init
+              case _ =>
+                paramsNoOuter
+            }
+
+            info = MethodType(newParams, clazz.tpe)
+        }
+      // Note: the info may be overwritten later with a generic signature
+      // parsed from SignatureATTR
+      sym.setInfo(if (jflags.isVarargs) arrayToRepeated(info) else info)
+      addParamNames(sym)
+    }
+  }
+
+  private def addParamNames(sym: Symbol): Unit = {
+    sym.getAndRemoveAttachment[ParamNames] match {
+      case Some(ParamNames(paramNames)) =>
+        if (sym.hasRawInfo && sym.isMethod) {
+          val params = sym.rawInfo.params
+          val paramNames1 = paramNames.takeRight(params.length)
+          (paramNames1 zip params).foreach {
+            case (nme.NO_NAME, _) => // param was ACC_SYNTHETIC; ignore
+            case (name, param) =>
+              param.resetFlag(SYNTHETIC)
+              param.name = name
+          }
+          if (isDeveloper && !sameLength(paramNames.toList, params)) {
+            // there's not anything we can do, but it's slightly worrisome
+            devWarning(
+              sm"""MethodParameters length mismatch while parsing $sym:
+                  |  rawInfo.params: ${sym.rawInfo.params}
+                  |  MethodParameters: ${paramNames.toList}""")
+          }
+        }
+      case _ =>
+    }
+  }
+
+  final class JavaGenericSignatureTypeCompleter(sig: String, isVarargs: Boolean) extends LazyType {
+    override def complete(sym: symbolTable.Symbol): Unit = {
+      var info = sigToType(sym, sig)
+      sym.setInfo(if (isVarargs) arrayToRepeated(info) else info)
```

Comment:
It seems this wasn't there before for the `sigToType` case, is this a bugfix?

Reply:
This used to be called in `parseMethod` after type was initially created based on the descriptor and after it was recreated based on the signature (if there was one). So it did apply in both cases.